tests/interface/TWLuxTests.cpp

```diff
@@ -0,0 +1,68 @@
+// file LICENSE at the root of the source code distribution tree.
+
+#include "TWTestUtilities.h"
+
+#include <TrustWalletCore/TWBech32Address.h>
+#include <TrustWalletCore/TWBitcoinAddress.h>
+#include <TrustWalletCore/TWBitcoinScript.h>
+#include <TrustWalletCore/TWHash.h>
+#include <TrustWalletCore/TWHDWallet.h>
+#include <TrustWalletCore/TWHRP.h>
+#include <TrustWalletCore/TWP2PKHPrefix.h>
+#include <TrustWalletCore/TWP2SHPrefix.h>
+#include <TrustWalletCore/TWPrivateKey.h>
+
+#include <gtest/gtest.h>
+
+TEST(Lux, Address) {
+    auto privateKey = WRAP(TWPrivateKey, TWPrivateKeyCreateWithData(DATA("a22ddec5c567b4488bb00f69b6146c50da2ee883e2c096db098726394d585730").get()));
+    auto publicKey = TWPrivateKeyGetPublicKeySecp256k1(privateKey.get(), true);
+    auto address = TWBitcoinAddress();
+    TWBitcoinAddressInitWithPublicKey(&address, publicKey, TWP2PKHPrefixLux);
+    auto addressString = WRAPS(TWBitcoinAddressDescription(address));
+    assertStringsEqual(addressString, "LV7LV7Z4bWDEjYkfx9dQo6k6RjGbXsg6hS");
+}
+
+TEST(Lux, ExtendedKeys) {
+    auto wallet = WRAP(TWHDWallet, TWHDWalletCreateWithMnemonic(
+        STRING("ripple scissors kick mammal hire column oak again sun offer wealth tomorrow wagon turn fatal").get(),
+        STRING("TREZOR").get()
+    ));
+
+    auto xpub = WRAPS(TWHDWalletGetExtendedPublicKey(wallet.get(), TWPurposeBIP44, TWCoinTypeLux, TWHDVersionXPUB));
+    auto xprv = WRAPS(TWHDWalletGetExtendedPrivateKey(wallet.get(), TWPurposeBIP44, TWCoinTypeLux, TWHDVersionXPRV));
+
+    assertStringsEqual(xpub, "xpub6CPmhSbYFoVy6rr2eGGkK7nbJYTL8SoWZYs3uSuWRpUg1s9Ln3xmEjMyycjvsF8qF76UEqh4yvmdCBbtXPKatMdrvo5bdhchvQqEZw4anDC");
+    assertStringsEqual(xprv, "xprv9yQRHw4eRRwftNmZYEjjwyqrkWcqiz5fCKwT74VtsUwh94pCEWeWgw3W8LEC1BVF5r6bJLuCnD6HR2X8bETk3mw5VBgVDuw2BNL2NAVGp5y");
+}
+
+TEST(Lux, DeriveFromXpub) {
+    auto xpub = STRING("xpub6Cb8Q6pDeS8PdKNbDv9Hvq4WpJXL3JvKvmHHwR1wD2H543hiCUE1f1tB5AXE6yg13k7xZ6PzEXMNUFHXk6kkx4RYte8VB1i4tCX9rwQVR4a");
+    auto pubKey3 = TWHDWalletGetPublicKeyFromExtended(xpub.get(), TWCurveSECP256k1, TWHDVersionXPUB, TWHDVersionXPRV, 0, 3);
+    auto pubKey5 = TWHDWalletGetPublicKeyFromExtended(xpub.get(), TWCurveSECP256k1, TWHDVersionXPUB, TWHDVersionXPRV, 0, 5);
+
+    TWBitcoinAddress address3;
+    TWBitcoinAddressInitWithPublicKey(&address3, pubKey3, TWP2PKHPrefixLux);
+    auto address3String = WRAPS(TWBitcoinAddressDescription(address3));
+
+    TWBitcoinAddress address5;
+    TWBitcoinAddressInitWithPublicKey(&address5, pubKey5, TWP2PKHPrefixLux);
+    auto address5String = WRAPS(TWBitcoinAddressDescription(address5));
+
+    assertStringsEqual(address3String, "LfJVQc6oq2T19WQLHAvMeHUjZVsspKFQu4");
+    assertStringsEqual(address5String, "LdEXFwEC9qZTVbVJdhNzHmrQBmUmG4VxzL");
+}
+
+TEST(Lux, LockScripts) {
+    auto script2 = WRAP(TWBitcoinScript, TWBitcoinScriptBuildForAddress(STRING("a4YtT82mWWxHZhLmdx7e5aroW92dqJoRs3").get()));
```

Comment:
Is this test case correct? `a4YtT82mWWxHZhLmdx7e5aroW92dqJoRs3` is not a Lux address?

Reply:
Thanks for noticing that. Yeah I missed it